packages/run-protocol/src/runStake/runStake.js

```diff
@@ -0,0 +1,171 @@
+// @ts-check
+import { handleParamGovernance } from '@agoric/governance';
+import { E, Far } from '@endo/far';
+import { makeAttestationFacets } from './attestation.js';
+import { makeRunStakeParamManager } from './params.js';
+import { makeRunStakeKit } from './runStakeKit.js';
+import { makeRunStakeManager } from './runStakeManager';
+
+/**
+ * Provide loans on the basis of staked assets that earn rewards.
+ *
+ * In addition to brands and issuers for `Staked`, `RUN`, and attestation,
+ * terms of the contract include a periodic interest rate plus
+ * a fee proportional to the amount borrowed, as well as a ratio
+ * of funds to mint and loan per unit of staked asset.
+ * These terms are subject to change by the `Electorate`
+ * and `electionManager` terms.
+ *
+ * @typedef {{
+ *   electionManager: VoteOnParamChange,
+ *   main: {
+ *     MintingRatio: ParamRecord<'ratio'>,
+ *     InterestRate: ParamRecord<'ratio'>,
+ *     LoanFee: ParamRecord<'ratio'>,
+ *     Electorate: ParamRecord<'invitation'>,
+ *   },
+ * }} RunStakeGovernanceTerms
+ *
+ * As in vaultFactory, `timerService` provides the periodic signal to
+ * charge interest according to `chargingPeriod` and `recordingPeriod`.
+ *
+ * @typedef { RunStakeGovernanceTerms & {
+ *   timerService: TimerService,
+ *   chargingPeriod: bigint,
+ *   recordingPeriod: bigint,
+ *   lienAttestationName?: string,
+ * }} RunStakeTerms
+ *
+ * The public facet provides access to invitations to get a loan
+ * or to return an attestation in order to release a lien on staked assets.
+ *
+ * @typedef {{
+ *   makeLoanInvitation: () => Promise<Invitation>,
+ *   makeReturnAttInvitation: () => Promise<Invitation>,
+ * }} RunStakePublic
+ *
+ * To take out a loan, get an `AttMaker` for your address from
+ * the creator of this contract, and use
+ * `E(attMaker).makeAttestation(stakedAmount)` to take out a lien
+ * and get a payment that attests to the lien. Provide the payment
+ * in the `Attestation` keyword of an offer,
+ * using `{ want: { RUN: amountWanted }}`.
+ *
+ * Then, using the invitationMakers pattern, use `AdjustBalances` to
+ * pay down the loan or otherwise adjust the `RUN` and `Attestation`.
+ *
+ * Finally, `Close` the loan, providing `{ give: RUN: debtAmount }}`
+ *
+ * To start the contract, authorize minting assets by providing `feeMintAccess`
+ * and provide access to the underlying staking infrastructure in `lienBridge`.
+ *
+ * @typedef {{
+ *   feeMintAccess: FeeMintAccess,
+ *   initialPoserInvitation: Invitation,
+ *   lienBridge: ERef<StakingAuthority>,
```

Comment:
I hope not. I thought about moving it from `privateArgs` to terms, but it's too powerful: it has `setLien`.

Reply:
With Electorate, we were able to separate invitation from invitationAmount. Since the clients of governed contracts very much want to know who can govern the contract, this was very important.  Similarly, the ContractGovernor inserts its own identity into the terms of the governed contract so clients can find and verify either from the other.

Can clients of a runStake contract be fooled into buying a pig in a poke if they can't identify the lienBridge? If clients would want to know, we could put the identity of the instance in the terms and ask zoe to verify the link between them.  Do we use a public or private facet of the lienBridge?

Reply:
`lienBridge` isn't a contract, so I don't think we can use Zoe instances to identify it.

I think the reason to trust a `runStake` instance is that you found it via `agoricNames`.